File: Difficulty: Basic/Replace all 0's with 5/replace-all-0s-with-5.cpp
/*you are required to complete this method*/
class Solution {
  public:
    int convertFive(int n) {
        // Your code here
        stack<int> st;
        if (n==0){
            return 5; 
        }
        while(n>0){
            int last = n%10;
            if(last == 0){
                st.push(5);
            }else{
                st.push(last);
            }
            n = n / 10;
        }
        
        int ans = 0;
        while(!st.empty()){
            ans = ans * 10 + st.top();
            st.pop();
        }
        return ans;
    }
};

File: Difficulty: Basic/Replace all 0's with 5/replace-all-0s-with-5.java
class Solution {
    public int convertFive(int n) {
        // code here
        
        if(n==0)
        {
            return 5;
        }
        
        int rev = 0;
        
        while(n!=0)
        {
            int last = n%10;
            if(last==0)
            {
                rev = rev * 10 + 5;
            }
            else
            {
                rev = rev * 10 + last;
            }
            n=n/10;
        }
        
        int ans = 0;
        
        while(rev!=0)
        {
            int l = rev%10;
            ans = ans * 10 + l;
            rev = rev/10;
        }
        
        return ans;
    }
}